website/content/en/pre-docs/AWS/constraints.md

```diff
@@ -0,0 +1,159 @@
+
+It is required, and specified by name. A suitable `KarpenterNodeRole` is created in the getting started guide.
+
+```
+spec:
+  provider:
+    instanceProfile: MyInstanceProfile
+```
+
+### LaunchTemplate
+
+A launch template is a set of configuration values sufficent for launching an EC2 instance (e.g., AMI, storage spec).
+
+A custom launch templay *may optionally* be specified by name. If none is specified, Karpenter will automatically create a launch template.
+
+Review the [Launch Template documentation](launch-templates.md) to learn how to create a custom one.
+
+```
+spec:
+  provider:
+    launchTemplate: MyLaunchTemplate
+```
+
+### SubnetSelector
+By default, Karpenter discovers subnets by tags. Alternatively, cluster subnets may be directly enumerated. 
+
+Subnets may be specified by AWS tag, or by name. Either approach supports wildcards. 
+
+Each instance gets *one* subnet *chosen* from this list.
+
+**Examples**
+
+Select all subnets with a specified tag:
+```
+spec:
+  provider:
+    subnetSelector:
+      kubernetes.io/cluster/MyCluster: '*'
+```
+
+Select subnets by name:
+```
+  subnetSelector:
+    Name: subnet-0fcd7006b3754e95e
+```
+
+Select subnets by an arbitary AWS tag key/value pair:
+```
+  subnetSelector:
+    MySubnetTag: value
+
+Select subnets using wildcards:
+```
+  subnetSelector:
+    Name: *public* 
+    MySubnetTag: '' # all resources with this tag
+
+```
+
+### SecurityGroupSelector
+
+Karpenter uses the EKS default security group, unless another is specified. The security group of an instance is comperable to a set of firewall rules.
+
+EKS creates at least two security groups, review the documentation for more info.
+
+Security Groups may be specified by AWS tag, or by name. Either approach supports wildcards. 
+
+Each instance gets *all* of the listed security groups.
+
+**Examples**
+
+Select all security groups with a specified tag:
+```
+spec:
+  provider:
+    securityGroupSelector:
+      kubernetes.io/cluster/MyKarpenterSecurityGroups: '*'
+```
+
+Select security groups by name, or another tag:
+```
+ securityGroupSSelector:
+   Name: sg-01077157b7cf4f5a8
+   MySecurityTag: '' # matches all resources with the tag
+```
+
+Select security groups by name using a wildcard:
+```
+ subnetSelector:
+   Name: *public*
+```
+
+### Tags
+
+All listed tags will be added to every node created by this provisioner.
+
+```
+spec:
+  provider:
+    tags:
+      InternalAccountingTag: 1234
+      dev.corp.net/app: Calculator
+      dev.corp.net/team: MyTeam
+```
+
+## Other Resources
+
+### Accelerators, GPU
+
+Accelerator (e.g., GPU) values include
+- `nvidia.com/gpu`
+- `amd.com/gpu`
+- `aws.amazon.com/neuron`
+
+Karpenter supports accelerators, such as GPUs.
+
+To enable instances with accelerators, use the [instance type
+well known label selector](#instance-types).
+
+Additionally, include a resource requirement in the workload manifest. Thus,

```

Comment:
Can you replace "Thus" with "This will cause...to be scheduled onto the appropriate node"?